### ISSUE TITLE
putting binary HGR images in a disk image does not always work as expected

### ISSUE THREAD

**Issue by fanfani** [paleotronic/diskm8#5]
hello again,

i have a bunch of HGR files i want to put in a disk image.

putting some of this images in a disk image results in corrupt files of a 1 block size:

```
(c) 2015 - 2018 Paleotronic.com

dsk:0:<no mount>:[       ]> mount blank.do

mount disk in slot 0

dsk:0:blank.do:> put gmm_hgr/G05

Backed up disk to: blank.do.20180407170539
Updated disk blank.do

dsk:0:blank.do:> put gmm_hgr/G06

Backed up disk to: blank.do.20180407170541
Updated disk blank.do

dsk:0:blank.do:> put gmm_hgr/G07

Backed up disk to:  blank.do.20180407170543
Updated disk  blank.do

dsk:0:blank.do:> put gmm_hgr/G08

Backed up disk to:  blank.do.20180407170544
Updated disk  blank.do

dsk:0:blank.do:> put gmm_hgr/G09

Backed up disk to:  blank.do.20180407170546
Updated disk  blank.do

dsk:0:blank.do:> cat *

NAME                               BLOCKS  RO  KIND                     ADDITONAL
g05                                    33      Binary File              (A$2A55)
g06                                    33      Binary File              (A$8080)
g07                                    33      Binary File              (A$D5AA)
g08                                     1      Binary File              (A$2A61)
g09                                     1      Binary File              

USED: 233                  FREE: 327                 

dsk:0:blank.do:>  
```

g08 and g09 are __not__ corrupt, since:

- if i put them them in the disk image using a similar tool (applecommander) they all have the correct size and load properly

- they all get correctly converted to BMP using a2pix, 

my guess is that some bytes in the beginning of the files are mistaken by EOF in diskm8 (or something like that)

....can you please check? 

ty :-)

[hgr_files.zip](https://github.com/paleotronic/diskm8/files/1886478/hgr_files.zip)

**Comment by undefinedopcode** [comment]
Thanks, 

You spotted an interesting bug related to the file size and prefix handling of AppleDOS images.  Binary files are supposed to have the load address and length encoded on them.  Address was being encoded but length was missed and was whatever data was at the start of the file.

I have fixed this and made it consistent.

Also, I've added a handy helper to the put command to allow specifying the load address: -

put FILENAME,A$XXXX,L$XXXX will allow setting the load address and length on files (similar to BSAVE).
